### PR TITLE
docs: add documentation and script for local docs dev environment

### PR DESCRIPTION
**Pull request by bencodezen**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [x] Other... Please describe: angular.dev infrastructure script change

## What is the current behavior?

There is no current path for building angular.dev locally.

Issue Number: N/A

## What is the new behavior?

Users can build a local dev environment of the Angular.dev environment.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by mgechev — APPROVED** [review]

**Comment by pkozlowski-opensource** [comment]
@bencodezen looks like this needs commit message update (see the failing lint job)

**Comment by bencodezen** [comment]
> @bencodezen looks like this needs commit message update (see the failing lint job)

Just saw that too. Squashing the commits to avoid the conflict. Thanks for the ping!

File: package.json [hunk 1]
```diff
@@ -43,6 +43,7 @@
     "devtools:build:chrome": "bazelisk build --config snapshot-build --//devtools/projects/shell-browser/src:flag_browser=chrome -- devtools/projects/shell-browser/src:prodapp",
     "devtools:build:firefox": "bazelisk build --config snapshot-build --//devtools/projects/shell-browser/src:flag_browser=firefox -- devtools/projects/shell-browser/src:prodapp",
     "devtools:test": "bazelisk test --config snapshot-build --//devtools/projects/shell-browser/src:flag_browser=chrome -- //devtools/...",
+    "docs": "yarn bazel run --config=aio_local_deps  //adev:serve",
```
Comment (devversion) [inline package.json:46]: This will not watch for changes as far as I understand. That would need to be ibazel then.

**Review by devversion — APPROVED** [review]

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit c4853e952d78d7f79d33503c2327e1a6cc0eb9b9.

**Pull request closed without merging**